### PR TITLE
Separate script arguments from the script file for InteractiveCompiler tasks

### PR DESCRIPTION
**Pull request by JoeRobich**
Fixes https://github.com/dotnet/roslyn/issues/17034

/CC  @tmat

### CODE REVIEW

File: src/Compilers/Core/MSBuildTask/CommandLineBuilderExtension.cs [hunk 1]
```diff
@@ -263,5 +263,14 @@ internal void AppendSwitchIfNotNull
                 }
             }
         }
+
+        internal void AppendArgumentsIfNotNull(string arguments)
```
Comment (tmat) [inline src/Compilers/Core/MSBuildTask/CommandLineBuilderExtension.cs]: >internal void AppendArgumentsIfNotNull(string arguments) [](start = 8, length = 56)

This is just used in one specific caller, so I'd inline.

File: src/Compilers/Core/MSBuildTask/CommandLineBuilderExtension.cs [hunk 2]
```diff
@@ -263,5 +263,14 @@ internal void AppendSwitchIfNotNull
...
+        {
+            if (!string.IsNullOrEmpty(arguments))
+            {
+                AppendSpaceIfNotEmpty();
+                AppendTextUnquoted(arguments);
```
Comment (tmat) [inline src/Compilers/Core/MSBuildTask/CommandLineBuilderExtension.cs]: >AppendTextUnquoted [](start = 16, length = 18)

It's odd that we expect the arguments to be already quoted. Wouldn't it make more sense to quote them as we are adding them to the command line?

File: src/Compilers/Core/MSBuildTask/CommandLineBuilderExtension.cs (hunk 1 above)
Comment (JoeRobich) [reply src/Compilers/Core/MSBuildTask/CommandLineBuilderExtension.cs]: `AppendSpaceIfNotEmpty()`  is a protected method so even if I inline this in the `InteractiveCompiler` I would still have to add method to expose that.

File: src/Compilers/Core/MSBuildTask/CommandLineBuilderExtension.cs (hunk 2 above)
Comment (JoeRobich) [reply src/Compilers/Core/MSBuildTask/CommandLineBuilderExtension.cs]: Updated to add as quoted text.

File: src/Compilers/Core/MSBuildTask/CommandLineBuilderExtension.cs (hunk 1 above)
Comment (tmat) [reply src/Compilers/Core/MSBuildTask/CommandLineBuilderExtension.cs]: I see. OK.

---
In reply to: [285295404](https://github.com/dotnet/roslyn/pull/35734#discussion_r285295404) [](ancestors = 285295404)

**Comment by JoeRobich** [comment]
Two legs of the integration test were never queued over the weekend. Canceling and retrying the legs did not kick off a run. The changes I made have no associated integration tests and should not affect integration tests. Merging

**Pull request merged**